### DISCUSSION TITLE
using of personal signature for meta transactions

### DISCUSSION THREAD

**Discussion by dawksh** [rainbow-me/rainbowkit#575] [Q&A] [answered]
I was wondering how to use meta transactions using biconomy mexa SDK, it requires a provider with signer but in rainbowkit under the hood wagmi provides useProvider hook which doesn't have any signer. Any ideas how do I go about it?

**Comment by saihaj** [comment]
You can:
- use [`useSigner`](https://wagmi.sh/docs/hooks/useSigner)
- use [`useAccount`](https://wagmi.sh/docs/hooks/useAccount) and do `connector?.getSigner()`

**Comment by dawksh** [comment]
my connector seems to be `undefined` for some reason

**Reply by jxom** [reply]
Are you connected to a wallet?

**Reply by dawksh** [reply]
yes I am

**Comment by dawksh** [comment, marked as answer]
I actually got it working, 
```
const { data: signer } = useSigner();
```

now this signer has a provider which is connected to the wallet and can be used to initialize Biconomy. it can be simply referenced by `signer.provider`
